src/main/java/com/ericsson/gerrit/plugins/eiffel/storage/EventStorage.java

```diff
@@ -0,0 +1,147 @@
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.ericsson.gerrit.plugins.eiffel.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.ConnectException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.sql.SQLException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.NoSuchElementException;
+import com.ericsson.gerrit.plugins.eiffel.handlers.DataBaseHandler;
+import com.ericsson.gerrit.plugins.eiffel.handlers.Table;
+
+public abstract class EventStorage {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(EventStorage.class);
+    protected static final String FILE_ENDING = "db";
+
+    protected File pluginDir;
+
+    public EventStorage(File pluginDir) {
+        this.pluginDir = pluginDir;
+    }
+
+    public abstract String getEventId(String project, String tableColumnName) throws NoSuchElementException, ConnectException, FileNotFoundException;
+
+    public abstract void saveEventId(String eiffelEventId, EiffelEvent eiffelEvent)
+            throws NoSuchElementException, SQLException, ConnectException;
+
+    protected String getLastSubmittedEiffelEvent(String project, String tableColumnName,
+            Table tableName)
+            throws NoSuchElementException, FileNotFoundException, ConnectException {
+        try {
+            File parentDir = new File(buildParentFilePath(pluginDir, project));
+            if (!(parentDir.exists())) {
+                throw new FileNotFoundException(parentDir.getName() + " could not be created.");
+            }
+
+            return getEventId(project, tableColumnName, tableName);
+        } catch (NoSuchElementException e) {
+            throw new NoSuchElementException(
+                    "Database did not return any value for this query\n" + "Exception Message:" + e.getMessage());
+        }
+    }
+
+    protected void setLastSubmittedEiffelEvent(String project, String tableColumnName,
+            String eiffelEvent,
+            Table tableName) throws NoSuchElementException, ConnectException, SQLException {
+        String fileName = String.format("%s.%s", project, FILE_ENDING);
+        DataBaseHandler dBHandler = new DataBaseHandler(pluginDir, fileName);
+        String parentPath = buildParentFilePath(pluginDir, project);
+        createParentDirsIfNecessary(parentPath);
```

Comment:
You need to explain why you need the parent file path because now it looks you are just creating them without using them.

Reply:
This file is mostly taken from the old code and i actually don't understand this quite, i will check :)

Reply:
Parent file path is when a project has parents, /somepath/baseProjectDir/children

It creates the dirs in local server disk so the DB file is saved in an existing directory, maybe this should be in the DBHandler. DBHandler will faiil to save the db file if the dirst does not exist.

Reply:
Sound like this should be moved then